gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -89,28 +92,57 @@ public Void call() throws IOException, ClassNotFoundException, InterruptedExcept
     Configuration conf = HadoopUtils.getConfFromState(this.jobProps);
     DateTime jobStartTime = new DateTime(DateTimeZone.forID(this.jobProps.getProp(
         ConfigurationKeys.COMPACTION_TIMEZONE, ConfigurationKeys.DEFAULT_COMPACTION_TIMEZONE)));
+
     if (this.jobProps.getPropAsBoolean(ConfigurationKeys.COMPACTION_JOB_LATE_DATA_MOVEMENT_TASK, false)) {
-      this.copyLateDataFiles(new Path(this.outputPath, "late"), conf);
+      List<Path> lateFilePaths = Lists.newArrayList();
+      for (String filePathString : this.jobProps.getPropAsList(ConfigurationKeys.COMPACTION_JOB_LATE_DATA_FILES)) {
+        if (FilenameUtils.isExtension(filePathString, getApplicableFileExtensions())) {
+          lateFilePaths.add(new Path(filePathString));
+        }
+      }
+      this.copyDataFiles(
+          new Path(this.outputPath, ConfigurationKeys.COMPACTION_LATE_FILES_DIRECTORY), lateFilePaths, conf);
```

Comment:
If deduplication = false, late files should not be copied to the output dir instead of "late" dir.